scipy/sparse/coo.py

```diff
@@ -278,31 +275,26 @@ def tocsc(self, copy=False):
             return csc_matrix(self.shape, dtype=self.dtype)
         else:
             M,N = self.shape
+            self.sum_duplicates()
             idx_dtype = get_index_dtype((self.col, self.row),
                                         maxval=max(self.nnz, M))
+            row = self.row.astype(idx_dtype, copy=False)
+            col = self.col.astype(idx_dtype, copy=False)
```

Comment:
Without these casts, a bunch of tests fail with: `ValueError: Output dtype not compatible with inputs.`. Might be a bug with sparsetools?